plugins/nodes/src/js/components/NodesGridView.tsx

```diff
@@ -1,49 +1,52 @@
+import * as React from "react";
 import { Trans } from "@lingui/macro";
 import classNames from "classnames";
-import PureRender from "react-addons-pure-render-mixin";
-import PropTypes from "prop-types";
-import React from "react";
 
 import Loader from "#SRC/js/components/Loader";
 import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
 import NodesList from "#SRC/js/structs/NodesList";
 
 import NodesGridDials from "./NodesGridDials";
 
-var MAX_SERVICES_TO_SHOW = 32;
-var OTHER_SERVICES_COLOR = 32;
-
-var NodesGridView = React.createClass({
-  displayName: "NodesGridView",
-
-  mixins: [PureRender],
-
-  propTypes: {
-    hasLoadingError: PropTypes.bool,
-    hiddenServices: PropTypes.array,
-    hosts: PropTypes.instanceOf(NodesList).isRequired,
-    receivedEmptyMesosState: PropTypes.bool,
-    receivedNodeHealthResponse: PropTypes.bool,
-    resourcesByFramework: PropTypes.object.isRequired,
-    selectedResource: PropTypes.string.isRequired,
-    serviceColors: PropTypes.object.isRequired,
-    services: PropTypes.array.isRequired
-  },
-
-  defaultProps: {
+const MAX_SERVICES_TO_SHOW = 32;
+const OTHER_SERVICES_COLOR = 32;
+
+interface ServiceColors {
+  [index: string]: number;
+}
+
+interface ResourcesByFramework {
+  [index: string]: object;
```

Comment:
It would be cool if we could really take advantage of types here and define what each resource object would look like but that's starting to smell like scope creep....